### PR TITLE
docs: update documents related to Plan Cache

### PR DESCRIPTION
**Pull request by qw4990**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: sql-prepared-plan-cache.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ key 中任何一项变动（如切换数据库，重命名 `Prepare` 语句，
 - 考虑到不同 `Execute` 的参数会不同，执行计划缓存为了保证适配性会禁止一些和具体参数值密切相关的激进查询优化手段，导致对特定的一些参数值，查询计划可能不是最优。比如查询的过滤条件为 `where a > ? and a < ?`，第一次 `Execute` 时参数分别为 2 和 1，考虑到这两个参数下次执行时可能会是 1 和 2，优化器不会生成对当前参数最优的 `TableDual` 执行计划。
 - 如果不考虑缓存失效和淘汰，一份执行计划缓存会对应各种不同的参数取值，理论上也会导致某些取值下执行计划非最优。比如查询过滤条件为 `where a < ?`，假如第一次执行 `Execute` 时用的参数值为 1，此时优化器生成最优的 `IndexScan` 执行计划放入缓存，在后续执行 `Exeucte` 时参数变为 10000，此时 `TableScan` 可能才是更优执行计划，但由于执行计划缓存，执行时还是会使用先前生成的 `IndexScan`。因此执行计划缓存更适用于查询较为简单（查询编译耗时占比较高）且执行计划较为固定的业务场景。
 
-目前执行计划缓存功能默认关闭，可以通过打开配置文件中 [`prepared-plan-cache` 项](/tidb-configuration-file.md#prepared-plan-cache)启用这项功能。
+目前执行计划缓存功能默认打开，可以通过变量 `tidb_enable_prepared_plan_cache` 启用这项功能。
```
Comment (Reminiscent) [inline sql-prepared-plan-cache.md:56]:
```suggestion
目前执行计划缓存功能默认打开，可以通过变量 `tidb_enable_prepared_plan_cache` 启用或关闭这项功能。
```

File: system-variables.md [hunk 2]
```diff
@@ -1466,6 +1475,25 @@ explain select * from t where age=5;
...
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值：`0.1`
+- 范围：`[0, 1]`
+- 这个变量用来控制 Prepared Plan Cache 触发内存保护机制的阈值，具体可见 [Prepared Plan Cache 的内存管理](sql-prepared-plan-cache.md#%E6%89%8B%E5%8A%A8%E6%B8%85%E7%A9%BA%E8%AE%A1%E5%88%92%E7%BC%93%E5%AD%98)。
+- 在 `v6.1.0` 之前这个开关通过配置文件进行配置，在升级上来时会自动继承配置文件的设置。
+
+### `tidb_prepared_plan_cache_size` <span class="version-mark">从 v6.1.0 版本开始引入</span>
+
+- 作用域：GLOBAL
```
Comment (Reminiscent) [inline system-variables.md:1490]: 这个没 session 级别的作用域么

Comment (qw4990) [reply system-variables.md:1490]: 目前没有，只有 global；

**Review by Reminiscent — APPROVED** [review]

File: tidb-configuration-file.md [hunk 3]
```diff
@@ -465,28 +465,6 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
-+ 类型为 uint，小于 0 的值会被转化为大整数。
-
-### `memory-guard-ratio`
-
-+ 用于防止 prepare plan cache 的内存用量超过 performance.server-memory-quota。当 prepare plan cache 的内存用量超过 server-memory-quota * (1 - prepared-plan-cache.memory-guard-ratio) 时，TiDB 会剔除 LRU 中的元素。
-+ 默认值：0.1
-+ 最小值：0
-+ 最大值：1
-
 ## opentracing
```
Comment (ran-huang) [inline tidb-configuration-file.md:468]: 这些是不是也要删掉啊
`enable-batch-dml`
`security.require-secure-transport` 
`oom-action`
`prepared-plan-cache.enabled`
`prepared-plan-cache.capacity`
`mem-quota-query`
`committer-concurrency`
`query-log-max-len`
`run-auto-analyze`

Comment (chrysan) [reply tidb-configuration-file.md:468]: 这个找 sql-infra 团队对齐一下吧，不属于这个 PR 的范畴。

**Review by chrysan — APPROVED** [review]

File: system-variables.md [hunk 4]
```diff
@@ -1466,6 +1475,25 @@ explain select * from t where age=5;
...
+- 是否持久化到集群：是
+- 默认值：`100`
+- 范围：`[0, 2147483647]`
```
Comment (ran-huang) [inline system-variables.md:1493]: morgan 提了一个 PR 里 [Range: `[1, 100000]`](https://github.com/pingcap/docs/pull/8689/files#diff-ec2cbb115cc58fe3afe2862eb5148732d323cbf7ae35dea97bd381a1f8d0e8a0R1651)，麻烦确认下哪边是正确的

File: system-variables.md [hunk 5]
```diff
@@ -801,6 +801,15 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 - 默认值：0
 - 这个变量用于控制是否开启 Apply 算子并发，并发数由 `tidb_executor_concurrency` 变量控制。Apply 算子用来处理关联子查询且默认无并发，所以执行速度较慢。打开 Apply 并发开关可增加并发度，提高执行速度。目前默认关闭。
 
+### `tidb_enable_prepared_plan_cache` <span class="version-mark">从 v6.1.0 版本开始引入</span>
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值：`ON`
+- 这个变量用来控制是否打开 Prepared Plan Cache。
+- 打开开关后会启动 [Prepared Plan Cache](sql-prepared-plan-cache.md) 功能，对 `Prepare` / `Execute` 请求的执行计划会进行缓存，以便在后续执行时跳过查询计划优化这个步骤，获得性能上的提升。
```
Comment (ran-huang) [inline system-variables.md:810]:
```suggestion
- 这个变量用来控制是否开启 [Prepared Plan Cache](/sql-prepared-plan-cache.md)。开启后，对 `Prepare`、`Execute` 请求的执行计划会进行缓存，以便在后续执行时跳过查询计划优化这个步骤，获得性能上的提升。
```

File: system-variables.md [hunk 6]
```diff
@@ -801,6 +801,15 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
+- 在 `v6.1.0` 之前这个开关通过配置文件进行配置，在升级上来时会自动继承配置文件的设置。
```
Comment (ran-huang) [inline system-variables.md:811]:
```suggestion
- 在 v6.1.0 之前这个开关通过 TiDB 配置文件 (`prepared-plan-cache.enabled`) 进行配置，升级到 v6.1.0 时会自动继承原有设置。
```

File: system-variables.md [hunk 7]
```diff
@@ -1466,6 +1475,25 @@ explain select * from t where age=5;
...
 - 这个变量用来控制是否在 profile 输出中标记出对应的 SQL 语句，用于定位和排查性能问题。
 
+
+### `tidb_prepared_plan_cache_memory_guard_ratio` <span class="version-mark">从 v6.1.0 版本开始引入</span>
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值：`0.1`
+- 范围：`[0, 1]`
+- 这个变量用来控制 Prepared Plan Cache 触发内存保护机制的阈值，具体可见 [Prepared Plan Cache 的内存管理](sql-prepared-plan-cache.md#%E6%89%8B%E5%8A%A8%E6%B8%85%E7%A9%BA%E8%AE%A1%E5%88%92%E7%BC%93%E5%AD%98)。
```
Comment (ran-huang) [inline system-variables.md:1485]:
```suggestion
- 这个变量用来控制 Prepared Plan Cache 触发内存保护机制的阈值，具体可见 [Prepared Plan Cache 的内存管理](/sql-prepared-plan-cache.md#prepared-plan-cache-的内存管理)。
```

File: system-variables.md [hunk 8]
```diff
@@ -1466,6 +1475,25 @@ explain select * from t where age=5;
...
+- 在 `v6.1.0` 之前这个开关通过配置文件进行配置，在升级上来时会自动继承配置文件的设置。
```
Comment (ran-huang) [inline system-variables.md:1486]:
```suggestion
- 在 v6.1.0 之前这个开关通过 TiDB 配置文件 (`prepared-plan-cache.memory-guard-ratio`) 进行配置，升级到 v6.1.0 时会自动继承原有设置。
```

File: system-variables.md [hunk 9]
```diff
@@ -1466,6 +1475,25 @@ explain select * from t where age=5;
...
+
+### `tidb_prepared_plan_cache_size` <span class="version-mark">从 v6.1.0 版本开始引入</span>
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值：`100`
+- 范围：`[0, 2147483647]`
+- 这个变量用来控制单个 `SESSION` 的 Prepared Plan Cache 最多能够缓存的计划数量，具体可见 [Prepared Plan Cache 的内存管理](sql-prepared-plan-cache.md#%E6%89%8B%E5%8A%A8%E6%B8%85%E7%A9%BA%E8%AE%A1%E5%88%92%E7%BC%93%E5%AD%98)。
+- 在 `v6.1.0` 之前这个开关通过配置文件进行配置，在升级上来时会自动继承配置文件的设置。
```
Comment (ran-huang) [inline system-variables.md:1495]:
```suggestion
- 在 v6.1.0 之前这个开关通过 TiDB 配置文件 (`prepared-plan-cache.capacity`) 进行配置，升级到 v6.1.0 时会自动继承原有设置。
```

File: system-variables.md [hunk 10]
```diff
@@ -1466,6 +1475,25 @@ explain select * from t where age=5;
...
+- 这个变量用来控制 Prepared Plan Cache 触发内存保护机制的阈值，具体可见 [Prepared Plan Cache 的内存管理](sql-prepared-plan-cache.md#%E6%89%8B%E5%8A%A8%E6%B8%85%E7%A9%BA%E8%AE%A1%E5%88%92%E7%BC%93%E5%AD%98)。
+- 在 `v6.1.0` 之前这个开关通过配置文件进行配置，在升级上来时会自动继承配置文件的设置。
+
+### `tidb_prepared_plan_cache_size` <span class="version-mark">从 v6.1.0 版本开始引入</span>
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值：`100`
+- 范围：`[0, 2147483647]`
+- 这个变量用来控制单个 `SESSION` 的 Prepared Plan Cache 最多能够缓存的计划数量，具体可见 [Prepared Plan Cache 的内存管理](sql-prepared-plan-cache.md#%E6%89%8B%E5%8A%A8%E6%B8%85%E7%A9%BA%E8%AE%A1%E5%88%92%E7%BC%93%E5%AD%98)。
```
Comment (ran-huang) [inline system-variables.md:1494]:
```suggestion
- 这个变量用来控制单个 `SESSION` 的 Prepared Plan Cache 最多能够缓存的计划数量，具体可见 [Prepared Plan Cache 的内存管理](/sql-prepared-plan-cache.md#prepared-plan-cache-的内存管理)。
```

File: sql-prepared-plan-cache.md [hunk 11]
```diff
@@ -132,6 +132,23 @@ MySQL [test]> select @@last_plan_from_cache;
 1 row in set (0.00 sec)
 ```
 
+## Prepared Plan Cache 的内存管理
+
+使用 Prepared Plan Cache 会有一定的内存开销，在内部测试中，平均每个缓存计划会消耗 `100KiB` 内存，且目前 Plan Cache 是 `SESSION` 级别的，因此总内存消耗大致为 `SESSION 个数 * SESSION 平均缓存计划个数 * 100KiB`。
```
Comment (ran-huang) [inline sql-prepared-plan-cache.md:137]:
```suggestion
使用 Prepared Plan Cache 会有一定的内存开销，在内部测试中，平均每个缓存计划会消耗 100KiB 内存，且目前 Plan Cache 是 `SESSION` 级别的，因此总内存消耗大致为 `SESSION 个数 * SESSION 平均缓存计划个数 * 100KiB`。
```

File: sql-prepared-plan-cache.md [hunk 12]
```diff
@@ -132,6 +132,23 @@ MySQL [test]> select @@last_plan_from_cache;
...
+
+比如目前 TiDB 实例的 `SESSION` 并发数是 50，平均每个 `SESSION` 大致缓存 100 个计划，则总内存开销为 `50 * 100 * 100KiB` 约等于 `500mb`。
+
+目前可以通过变量 `tidb_prepared_plan_cache_size` 来设置每个 `SESSION` 最多缓存的计划数量,针对不同的环境，推荐的设置如下：
```
Comment (ran-huang) [inline sql-prepared-plan-cache.md:141]:
```suggestion
目前可以通过变量 `tidb_prepared_plan_cache_size` 来设置每个 `SESSION` 最多缓存的计划数量，针对不同的环境，推荐的设置如下：
```

File: sql-prepared-plan-cache.md [hunk 13]
```diff
@@ -132,6 +132,23 @@ MySQL [test]> select @@last_plan_from_cache;
...
+
+- MEM <= 64 GB; tidb_prepared_plan_cache_size = 50
+- MEM > 64 GB; tidb_prepared_plan_cache_size = 100
```
Comment (ran-huang) [inline sql-prepared-plan-cache.md:144]: MEM 是指部署 TiDB 的机器的内存吗？能否写更具体一点

File: sql-prepared-plan-cache.md [hunk 14]
```diff
@@ -132,6 +132,23 @@ MySQL [test]> select @@last_plan_from_cache;
...
+
+当 TiDB Server 的内存余量小于一定阈值时，会触发 Plan Cache 的内存保护机制，此时会对一些缓存的计划进行逐出。
+
+目前该阈值由变量 `tidb_prepared_plan_cache_memory_guard_ratio` 控制，默认为 0.1，即 10%，也就是当剩余内存不足 10%（使用内存超过 90%）时，会触发此机制。
+
+由于内存限制，Plan Cache 可能出现 Cache Miss 的情况，可以通过 `Plan Cache Miss OPS` 监控进行查看。
```
Comment (ran-huang) [inline sql-prepared-plan-cache.md:150]:
```suggestion
由于内存限制，Plan Cache 可能出现 Cache Miss 的情况，可以通过 Grafana 中的 [`Plan Cache Miss OPS` 监控](/grafana-tidb-dashboard.md)查看。
```

File: system-variables.md (hunk 4 above)
Comment (qw4990) [reply system-variables.md:1492]: 稍等我确认下

Comment (qw4990) [reply system-variables.md]: 应该为 100000，已经更新

File: sql-prepared-plan-cache.md (hunk 13 above)
Comment (qw4990) [reply sql-prepared-plan-cache.md]: 把 MEM 换成了 “TiDB Server 实例内存阈值”，麻烦再看下

File: sql-prepared-plan-cache.md [hunk 15]
```diff
@@ -53,7 +53,7 @@ key 中任何一项变动（如切换数据库，重命名 `Prepare` 语句，
 - 考虑到不同 `Execute` 的参数会不同，执行计划缓存为了保证适配性会禁止一些和具体参数值密切相关的激进查询优化手段，导致对特定的一些参数值，查询计划可能不是最优。比如查询的过滤条件为 `where a > ? and a < ?`，第一次 `Execute` 时参数分别为 2 和 1，考虑到这两个参数下次执行时可能会是 1 和 2，优化器不会生成对当前参数最优的 `TableDual` 执行计划。
 - 如果不考虑缓存失效和淘汰，一份执行计划缓存会对应各种不同的参数取值，理论上也会导致某些取值下执行计划非最优。比如查询过滤条件为 `where a < ?`，假如第一次执行 `Execute` 时用的参数值为 1，此时优化器生成最优的 `IndexScan` 执行计划放入缓存，在后续执行 `Exeucte` 时参数变为 10000，此时 `TableScan` 可能才是更优执行计划，但由于执行计划缓存，执行时还是会使用先前生成的 `IndexScan`。因此执行计划缓存更适用于查询较为简单（查询编译耗时占比较高）且执行计划较为固定的业务场景。
 
-目前执行计划缓存功能默认关闭，可以通过打开配置文件中 [`prepared-plan-cache` 项](/tidb-configuration-file.md#prepared-plan-cache)启用这项功能。
+目前执行计划缓存功能默认打开，可以通过变量 `tidb_enable_prepared_plan_cache` 启用或关闭这项功能。
```
Comment (ran-huang) [inline sql-prepared-plan-cache.md:56]:
```suggestion
目前执行计划缓存功能默认打开，可以通过变量 [`tidb_enable_prepared_plan_cache`](/system-variables.md#tidb_enable_prepared_plan_cache-从-v610-版本开始引入) 启用或关闭这项功能。
```

File: sql-prepared-plan-cache.md [hunk 16]
```diff
@@ -132,6 +132,23 @@ MySQL [test]> select @@last_plan_from_cache;
...
+## Prepared Plan Cache 的内存管理
+
+使用 Prepared Plan Cache 会有一定的内存开销，在内部测试中，平均每个缓存计划会消耗 100KiB 内存，且目前 Plan Cache 是 `SESSION` 级别的，因此总内存消耗大致为 `SESSION 个数 * SESSION 平均缓存计划个数 * 100KiB`。
+
+比如目前 TiDB 实例的 `SESSION` 并发数是 50，平均每个 `SESSION` 大致缓存 100 个计划，则总内存开销为 `50 * 100 * 100KiB` 约等于 `500mb`。
+
+目前可以通过变量 `tidb_prepared_plan_cache_size` 来设置每个 `SESSION` 最多缓存的计划数量，针对不同的环境，推荐的设置如下：
+
+- TiDB Server 实例内存阈值 <= 64 GB; tidb_prepared_plan_cache_size = 50
+- TiDB Server 实例内存阈值 > 64 GB; tidb_prepared_plan_cache_size = 100
```
Comment (ran-huang) [inline sql-prepared-plan-cache.md:144]:
```suggestion
- TiDB Server 实例内存阈值 <= 64 GB 时，`tidb_prepared_plan_cache_size = 50`
- TiDB Server 实例内存阈值 > 64 GB 时，`tidb_prepared_plan_cache_size = 100`
```

**Review by ran-huang — APPROVED** [review]

File: system-variables.md [hunk 17]
```diff
@@ -1466,6 +1474,25 @@ explain select * from t where age=5;
 - 范围：`[0, 1]`
 - 这个变量用来控制是否在 profile 输出中标记出对应的 SQL 语句，用于定位和排查性能问题。
 
+
```
Comment (Oreoxmt) [inline system-variables.md:1477]:
```suggestion
```

File: system-variables.md [hunk 18]
```diff
@@ -1466,6 +1474,24 @@ explain select * from t where age=5;
...
+- 范围：`[0, 1]`
+- 这个变量用来控制 Prepared Plan Cache 触发内存保护机制的阈值，具体可见 [Prepared Plan Cache 的内存管理](/sql-prepared-plan-cache.md#prepared-plan-cache-的内存管理)。
+- 在 v6.1.0 之前这个开关通过 TiDB 配置文件 (`prepared-plan-cache.memory-guard-ratio`) 进行配置，升级到 v6.1.0 时会自动继承原有设置。
+
+### `tidb_prepared_plan_cache_size` <span class="version-mark">从 v6.1.0 版本开始引入</span>
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值：`100`
+- 范围：`[0, 100000]`
```
Comment (ran-huang) [inline system-variables.md:1491]:
```suggestion
- 范围：`[1, 100000]`
```
https://github.com/pingcap/tidb/blob/dce5064e9e15d0b17c36f73e21fc71bcaa419e90/sessionctx/variable/sysvar.go#L767

File: sql-prepared-plan-cache.md [hunk 19]
```diff
@@ -132,6 +132,23 @@ MySQL [test]> select @@last_plan_from_cache;
 1 row in set (0.00 sec)
 ```
 
+## Prepared Plan Cache 的内存管理
+
+使用 Prepared Plan Cache 会有一定的内存开销，在内部测试中，平均每个缓存计划会消耗 100KiB 内存，且目前 Plan Cache 是 `SESSION` 级别的，因此总内存消耗大致为 `SESSION 个数 * SESSION 平均缓存计划个数 * 100KiB`。
```
Comment (Oreoxmt) [inline sql-prepared-plan-cache.md:137]:
```suggestion
使用 Prepared Plan Cache 会有一定的内存开销，在内部测试中，平均每个缓存计划会消耗 100 KiB 内存，且目前 Plan Cache 是 `SESSION` 级别的，因此总内存消耗大致为 `SESSION 个数 * SESSION 平均缓存计划个数 * 100KiB`。
```

File: sql-prepared-plan-cache.md [hunk 20]
```diff
@@ -132,6 +132,23 @@ MySQL [test]> select @@last_plan_from_cache;
...
+
+比如目前 TiDB 实例的 `SESSION` 并发数是 50，平均每个 `SESSION` 大致缓存 100 个计划，则总内存开销为 `50 * 100 * 100KiB` 约等于 `500mb`。
```
Comment (Oreoxmt) [inline sql-prepared-plan-cache.md:139]:
```suggestion
比如目前 TiDB 实例的 `SESSION` 并发数是 50，平均每个 `SESSION` 大致缓存 100 个计划，则总内存开销为 `50 * 100 * 100KiB` 约等于 `500MB`。
```

**Review by Oreoxmt — COMMENTED** [review]
Rest LGTM

File: sql-prepared-plan-cache.md (hunk 20 above)
Comment (Oreoxmt) [reply sql-prepared-plan-cache.md:139]: @qw4990 50 * 100 * 100KiB =  512 MB 需要将 500MB 修改为 512MB 吗

**Review by Oreoxmt — APPROVED** [review]

**Pull request merged**